pwndbg/commands/mmap.py

```diff
@@ -0,0 +1,239 @@
+MAP values: SHARED (1), PRIVATE (2), SHARED_VALIDATE (3), FIXED (0x10),
+            ANONYMOUS (0x20)
+
+Flags and protection values can be either a string containing the names of the
+flags or permissions or a single number corresponding to the bitwise OR of the
+protection and flag numbers.
+
+Examples:
+    mmap 0x0 4096 PROT_READ|PROT_WRITE|PROT_EXEC MAP_PRIVATE|MAP_ANONYMOUS -1 0
+     - Maps a new private+anonymous page with RWX permissions at a location
+       decided by the kernel.
+
+    mmap 0x0 4096 PROT_READ MAP_PRIVATE 10 0
+     - Maps 4096 bytes of the file pointed to by file descriptor number 10 with
+       read permission at a location decided by the kernel.
+
+    mmap 0xdeadbeef 0x1000
+     - Maps a new private+anonymous page with RWX permissions at a page boundary
+       near 0xdeadbeef.
+""",
+)
+parser.add_argument(
+    "addr", help="Address hint to be given to mmap.", type=pwndbg.commands.sloppy_gdb_parse
+)
+parser.add_argument(
+    "length",
+    help="Length of the mapping, in bytes. Needs to be greater than zero.",
+    type=int,
+)
+parser.add_argument(
+    "prot",
+    help='Prot enum or int as in mmap(2). Eg. "PROT_READ|PROT_EXEC" or 7 (for RWX).',
+    type=str,
+    nargs="?",
+    default="7",
+)
+parser.add_argument(
+    "flags",
+    help='Flags enum or int as in mmap(2). Eg. "MAP_PRIVATE|MAP_ANONYMOUS" or 0x22.',
+    type=str,
+    nargs="?",
+    default="0x22",
+)
+parser.add_argument(
+    "fd",
+    help="File descriptor of the file to be mapped, or -1 if using MAP_ANONYMOUS.",
+    type=int,
+    nargs="?",
+    default=-1,
+)
+parser.add_argument(
+    "offset",
+    help="Offset from the start of the file, in bytes, if using file based mapping.",
+    type=int,
+    nargs="?",
+    default=0,
+)
+parser.add_argument(
+    "--quiet", "-q", help="Disable address validity warnings and hints", action="store_true"
+)
+parser.add_argument(
+    "--force", "-f", help="Force potentially unsafe actions to happen", action="store_true"
+)
+
+
+prot_dict = {
+    "PROT_NONE": 0x0,
+    "PROT_READ": 0x1,
+    "PROT_WRITE": 0x2,
+    "PROT_EXEC": 0x4,
+}
+
+flag_dict = {
+    "MAP_SHARED": 0x1,
+    "MAP_PRIVATE": 0x2,
+    "MAP_SHARED_VALIDATE": 0x3,
+    "MAP_FIXED": 0x10,
+    "MAP_ANONYMOUS": 0x20,
+}
+
+
+def prot_str_to_val(protstr):
+    """Heuristic to convert PROT_EXEC|PROT_WRITE to integer value."""
+    prot_int = 0
+    for k, v in prot_dict.items():
+        if k in protstr:
+            prot_int |= v
+    return prot_int
+
+
+def flag_str_to_val(flagstr):
+    """Heuristic to convert MAP_SHARED|MAP_FIXED to integer value."""
+    flag_int = 0
+    for k, v in flag_dict.items():
+        if k in flagstr:
+            flag_int |= v
+    return flag_int
+
+
+def parse_str_or_int(val, parser):
+    """
+    Try parsing a string with one of the parsers above or by converting it to
+    an int, or passes the value through if it is already an integer.
+    """
+    if type(val) == str:
```

Comment:
```suggestion
    if type(val) is str:
```
Alternatively: `if isinstance(val, str):` which will support inheritance